tests/smoke-tests/src/test/java/org/apache/activemq/artemis/tests/smoke/paging/FloodServerWithAsyncSendTest.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.activemq.artemis.tests.util.CFUtil;
+import org.apache.activemq.artemis.utils.RandomUtil;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FloodServerWithAsyncSendTest extends SmokeTestBase {
+
+   public static final String SERVER_NAME_0 = "paging";
+
+   volatile boolean running = true;
+
+   AtomicInteger errors = new AtomicInteger(0);
+
+   @Before
+   public void before() throws Exception {
+      cleanupData(SERVER_NAME_0);
+      startServer(SERVER_NAME_0, 0, 30000);
+   }
+
+   @Test
+   public void testAsyncPagingOpenWire() throws Exception {
+      String protocol = "OPENWIRE";
+      internalTest(protocol);
+
+   }
+
+   ConnectionFactory newCF(String protocol) {
+      if (protocol.equalsIgnoreCase("OPENWIRE")) {
+         return CFUtil.createConnectionFactory(protocol, "tcp://localhost:61616?jms.useAsyncSend=true");
+      } else {
+         Assert.fail("unsuported protocol");
+         return null;
+      }
+   }
+
+   private void internalTest(String protocol) throws Exception {
+
+      Thread consume1 = new Thread(() -> consume(protocol, "queue1"), "ProducerQueue1");
+      consume1.start();
+      Thread consume2 = new Thread(() -> consume(protocol, "queue2"), "ProducerQueue2");
+      consume2.start();
+
+      Thread produce1 = new Thread(() -> produce(protocol, "queue1"), "ConsumerQueue1");
+      produce1.start();
+      Thread produce2 = new Thread(() -> produce(protocol, "queue2"), "ConsumerQueue2");
+      produce2.start();
+
+      Thread.sleep(10_000);
+
+      running = false;
+
+      consume1.join();
+      consume2.join();
+      produce1.join();
+      produce2.join();
+
+      ConnectionFactory factory = newCF("openwire");
+      Connection connection = factory.createConnection();
+      connection.start();
+      Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+      Queue queue = session.createQueue("queue3");
+      MessageConsumer consumer = session.createConsumer(queue);
+
+      MessageProducer producer = session.createProducer(queue);
+
+      String random = RandomUtil.randomString();
+
+      producer.send(session.createTextMessage(random));
+      TextMessage message = (TextMessage) consumer.receive(1000);
+      Assert.assertNotNull(message);
+      Assert.assertEquals(random, message.getText());
+      connection.close();
+
+      Assert.assertEquals(0, errors.get());
+
+   }
+
+   protected void consume(String protocol, String queueName) {
+      ConnectionFactory factory = newCF(protocol);
+      Connection connection = null;
+      try {
+         connection = factory.createConnection();
+         connection.start();
+         Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         Queue queue = session.createQueue(queueName);
+         MessageConsumer consumer = session.createConsumer(queue);
+         int rec = 0;
+         while (running) {
+            consumer.receive(5000);
```

Comment:
The point is that a third connection should still allowed to perform. The server should not become unresponsive which is what's happening on this situation.

My laptop has some bad cryptographic thing that slows the server down to a point where I can't guarantee the receives and senders will actually happen. I thought it would be safer to just check the unresponsive server than the actual producers and consumers.

the test is reproducing the issue as I expected here.